csharp-advanced-topics/MemoryInCSharp/Tests/BenchmarksLiveTest.cs

```diff
@@ -0,0 +1,22 @@
+using App;
+using BenchmarkDotNet.Reports;
+
+namespace Tests;
+public class BenchmarksLiveTest
+{
+    [Fact]
+    public void GivenMemoryAndStandardArrayAreInitialized_WhenElementsAreModifiedInBothArrays_ThenMemoryArrayModificationPerformanceIsBetter()
+    {
+        var summary = MemoryVsArrayBenchmarks.RunBenchmarks();
+        var standardArrayBenchmarkResult = GetBenchmarkResult(summary, nameof(MemoryVsArrayBenchmarks.ModifyStandardArrayElementsAsync));
+        var memoryArrayBenchmarkResult = GetBenchmarkResult(summary, nameof(MemoryVsArrayBenchmarks.ModifyMemoryArrayElementsAsync));
+        
+        Assert.True(memoryArrayBenchmarkResult < standardArrayBenchmarkResult);
+    }
+   private static double GetBenchmarkResult(Summary summary, string benchmarkName)
+    {
+        return summary.Reports
+            .First(r => r.BenchmarkCase.Descriptor.WorkloadMethod.Name == benchmarkName)
+            .ResultStatistics!.Mean;
+    }
```

Comment:
Our unit test should not be running a full benchmark of an array containing 10_000_000 elements. It would be better to rethink our methods so that we can pass in a smaller array and call our methods directly, executing them once and ensuring the behavior is as expected.

Reply:
I wanted to add a followup comment, because this is also a fragile unit test. When I ran it locally on my system, it failed. It also takes over a minute to run the unit test, which is too long.

Reply:
@jshergal, Thank you for your feedback. I'd like to clarify that the test in question is an integration test and not a unit test. As such, it has the prefix `Live`. These tests are not executed as part of the Jenkins Pipeline. With regard to the failure on your machine, it's worth noting that benchmark results can vary from one machine to another. This has already been communicated to CodeMaze.

Reply:
@goldytech - Noted regarding the live test. I missed that in the full method name and while I understand that, the execution time difference between the two methods is so minimal that it makes this a fragile test. Whether we refer to it as an integration test or not, it is still not a good test and does nothing to validate the correctness of our code. It would be better to have actual unit tests that show the correctness of the algorithm we have crafted. That is my point. An integration test with intermittent random failures is not a good test.